internal/service/ec2/vpc_ipam_preview_next_cidr.go

```diff
@@ -0,0 +1,125 @@
+package ec2
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	// "github.com/hashicorp/terraform-provider-aws/internal/flex"
+	// "github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceVPCIpamPreviewNextCidr() *schema.Resource {
+	return &schema.Resource{
+		Create: ResourceVPCIpamPreviewNextCidrCreate,
+		Read:   ResourceVPCIpamPreviewNextCidrRead,
+		Delete: schema.Noop,
+		Schema: map[string]*schema.Schema{
+			"cidr": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			// // temp comment out till bug is resolved
+			// "disallowed_cidrs": {
+			// 	Type:     schema.TypeSet,
+			// 	Optional: true,
+			// 	ForceNew: true,
+			// 	Elem: &schema.Schema{
+			// 		Type: schema.TypeString,
+			// 		ValidateFunc: validation.Any(
+			// 			verify.ValidIPv4CIDRNetworkAddress,
+			// 			// Follow the numbers used for netmask_length
+			// 			validation.IsCIDRNetwork(0, 32),
+			// 		),
+			// 	},
+			// },
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipam_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"netmask_length": {
+				// Possible netmask lengths for IPv4 addresses are 0 - 32.
+				// AllocateIpamPoolCidr API
+				//   - If there is no DefaultNetmaskLength allocation rule set on the pool,
+				//   you must specify either the NetmaskLength or the CIDR.
+				//   - If the DefaultNetmaskLength allocation rule is set on the pool,
+				//   you can specify either the NetmaskLength or the CIDR and the
+				//   DefaultNetmaskLength allocation rule will be ignored.
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 32),
+			},
+		},
+	}
+}
+
+func ResourceVPCIpamPreviewNextCidrCreate(d *schema.ResourceData, meta interface{}) error {
+	poolId := d.Get("ipam_pool_id").(string)
+	uniqueValue := resource.UniqueId()
+
+	// preview will not produce an IpamPoolAllocationId. Hence use the uniqueValue instead
+	d.SetId(encodeVPCIpamPreviewNextCidrID(uniqueValue, poolId))
+
+	return ResourceVPCIpamPreviewNextCidrRead(d, meta)
```

Comment:
```suggestion
	return resourceVPCIpamPreviewNextCidrRead(d, meta)
```

Reply:
Resolved in 7ac8f4bce5d6b827292ba98098259acd44f7991b